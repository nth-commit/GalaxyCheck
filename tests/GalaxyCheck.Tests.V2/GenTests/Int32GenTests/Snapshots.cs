using System.Collections.Generic;
using System.Linq;
using Xunit;
using GalaxyCheck;
using GalaxyCheck.Gens;
using Snapshooter.Xunit;
using Snapshooter;

namespace Tests.V2.GenTests.Int32GenTests
{
    public class Snapshots
    {
        [Fact]
        public void Values()
        {
            var seeds = Enumerable.Range(0, 5);

            var gens = new Dictionary<string, IInt32Gen>
            {
                { "Between_0_And_10", Gen.Int32().Between(0, 10) },
                { "Between_0_And_100", Gen.Int32().Between(0, 100) },
                { "Between_0_And_10000", Gen.Int32().Between(0, 10000) },
                { "Between_Default", Gen.Int32() },
            };

            var biases = new[] { Gen.Bias.None, Gen.Bias.WithSize };

            foreach (var seed in seeds)
            foreach (var (betweenDescription, gen) in gens)
            foreach (var bias in biases)
            {
                var sample = gen.WithBias(bias).Sample(seed: seed);

                var nameExtension = string.Join("_", new[]
                {
                    betweenDescription,
                    $"Bias_{bias}",
                    $"Seed_{seed}"
                });

                Snapshot.Match(sample, SnapshotNameExtension.Create(nameExtension));
            }
        }

        [Fact]
        public void ExampleSpaces()
        {
            var seeds = Enumerable.Range(0, 5);

            var gens = new Dictionary<string, IInt32Gen>
            {
                { "Between_0_And_10", Gen.Int32().Between(0, 10) },
                { "Between_0_And_100", Gen.Int32().Between(0, 100) },
            };

            var biases = new[] { Gen.Bias.None, Gen.Bias.WithSize };

            foreach (var seed in seeds)
            foreach (var (betweenDescription, gen) in gens)
            foreach (var bias in biases)
            {
                var sample = gen
                    .WithBias(bias)
                    .Advanced.SampleOneExampleSpace(seed: seed, size: 75)
                    .Take(500)
                    .Render(x => x.ToString());

                var nameExtension = string.Join("_", new[]
                {
                    betweenDescription,
                    $"Bias_{bias}",
                    $"Seed_{seed}"
                });

                Snapshot.Match(sample, SnapshotNameExtension.Create(nameExtension));
            }
        }
    }
}
