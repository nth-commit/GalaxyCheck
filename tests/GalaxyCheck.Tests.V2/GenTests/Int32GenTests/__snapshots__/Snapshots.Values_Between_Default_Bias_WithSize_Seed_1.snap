[
  0,
  -1,
  0,
  -1,
  1,
  -1,
  0,
  0,
  0,
  0,
  -1,
  -1,
  0,
  -1,
  1,
  1,
  -4,
  -4,
  -5,
  -1,
  -5,
  4,
  -3,
  1,
  3,
  -3,
  3,
  -10,
  5,
  2,
  -10,
  -4,
  -5,
  -1,
  7,
  9,
  -5,
  -1,
  -25,
  -22,
  -75,
  -86,
  -42,
  -7,
  31,
  16,
  3,
  -26,
  -47,
  64,
  -7716,
  -3788,
  -3244,
  -853,
  -6031,
  -3927,
  -819,
  -5108,
  9938,
  -6010,
  201,
  -799,
  -2147483648,
  -9927607,
  -77753319,
  -2147483648,
  -96704942,
  -15097598,
  -60159922,
  -16649421,
  -39198409,
  -83092779,
  -50123213,
  -51684908,
  -6127712,
  -2095232836,
  -1437248197,
  -842898323,
  -404042706,
  2147483647,
  -751612620,
  2147483647,
  -609896151,
  -166694624,
  -914392346,
  -1861121154,
  -2147483648,
  -1172508920,
  -808745091,
  -2147483648,
  -2147483648,
  2147483647,
  1876671735,
  -2102701749,
  -2147483648,
  537533351,
  -1691465726,
  -2147483648,
  -791478696,
  1513521051
]
